core/CronArchive.php

```diff
@@ -874,53 +877,61 @@ protected function processArchiveDays($idSite, $lastTimestampWebsiteProcessedDay
             return false;
         }
 
-        $content = $this->request($url);
-        $daysResponse = Common::safe_unserialize($content);
+        $visitsLastDays = 0;
 
-        if (empty($content)
-            || !is_array($daysResponse)
-            || count($daysResponse) == 0
-        ) {
-            // cancel marking the site as reprocessed
-            if ($websiteInvalidatedShouldReprocess) {
-                $store = new SitesToReprocessDistributedList();
-                $store->add($idSite);
-            }
+        list($visitsToday, $isThereArchive) = $this->isThereAValidArchiveForPeriod($idSite, 'day', $segment = '');
```

Comment:
@diosmosis wondering... in the method `isThereAValidArchiveForPeriod` do we always need to call `invalidateArchivedReportsForSitesThatNeedToBeArchivedAgain()` first in case there is a race condition? Like currently the archiver starts and calls `invalidateArchivedReportsForSitesThatNeedToBeArchivedAgain()`... the archiving may take many seconds, minutes, hours... by the time we get here there might be more `getRememberedArchivedReportsThatShouldBeInvalidated` because there were visits since the archiver started.

Reply:
👍 yes, makes sense